packages/argo-checkout/src/extension-points/api/post-purchase/post-purchase.ts

```diff
@@ -109,3 +102,104 @@ interface Metafield {
   value: string | number;
   valueType: 'integer' | 'string' | 'json_string';
 }
+
+interface ExplicitDiscount {
+  value: number;
+  valueType: ExplicitDiscountType;
+  title: string;
+}
+
+interface AddVariantChange {
+  type: ChangeType.AddVariant;
+  variantId: number;
+  quantity: number;
+  discount?: ExplicitDiscount;
+}
+
+type Change = AddVariantChange;
+
+interface Changeset {
+  changes: Change[];
+}
+
+interface Money {
+  amount: string;
+  /** In ISO 4217 format */
+  currencyCode: string;
+}
+
+/** Represents an amount in both shop and presentment currencies. */
+interface MoneyBag {
+  shopMoney: Money;
+  presentmentMoney: Money;
+}
+
+interface LineItem {
+  /** The discounted price per quantity. */
+  discountedPriceSet: MoneyBag;
+  /** The price per quantity */
+  priceSet: MoneyBag;
+}
+
+interface AddedTaxLine {
+  /** The tax amount */
+  priceSet: MoneyBag;
+  rate: string;
+  title: string;
+}
+
+interface UpdatedLineItem {
+  /** The discounted price per quantity. */
+  discountedPriceSet: MoneyBag;
+  /** The price per quantity */
+  priceSet: MoneyBag;
+  productId: number;
+  variantId: number;
+  /** How many items are being purchased in this line.*/
+  quantity: number;
+}
+
+interface UpdatedPurchase {
+  /** Updated total price of the purchase with discounts but before shipping, taxes, and tips. */
+  subtotalPriceSet: MoneyBag;
+  /** Updated final price of the purchase */
+  totalPriceSet: MoneyBag;
+  addedTaxLines: AddedTaxLine[];
+  updatedLineItems: UpdatedLineItem[];
+  /** The amount left unpaid after the update */
+  totalOutstandingSet: MoneyBag;
+}
+
+interface ChangesetError {
+  code: string;
+  message: string;
+}
+
+type ChangesetResult =
```

Comment:
This construct will make it so that if the status is `processed` or `partially_processed`, `updatedPurchase` is guaranteed to not be `null`

Is that assumption correct? cc @dotLou @mgingras

Reply:
@ginmrt ?

Reply:
haha, sorry about that. 🙇  I assume that happens quite often.

Reply:
> This construct will make it so that if the status is `processed` or `partially_processed`, `updatedPurchase` is guaranteed to not be `null`
> 
> Is that assumption correct?

That's my understanding, but I think @xiaopong-shopify and @ginmrt are the best placed to know for sure given they worked on the apply changeset API.

Reply:
That's correct 👍 🎉

Reply:
On the plus side (pun intended), by tagging @mgringas, you tapped into some great UX expertise.